libs/dex/src/main/java/io/crate/data/AsyncFlatMapBatchIterator.java

```diff
@@ -73,6 +73,10 @@ public boolean moveNext() {
                 }
                 return false;
             } else {
+                if (mappedElements == null) {
+                    // This is the case if a consumer didn't call loadNextBatch after a previous moveNext call returned false
```

Comment:
Isn't this also an issue that the consumer behaves like that?

Reply:
Yes, I  can look into that as well but I think the `AsyncFlatMapBatchIterator` should also be more resilient and cope with such an access pattern